tokio-io/src/io/async_write_ext.rs

```diff
@@ -37,6 +38,28 @@ pub trait AsyncWriteExt: AsyncWrite {
     {
         shutdown(self)
     }
+
+    /// Wraps the writer in a [`BufWriter`] so that small writes are batched to the underlying
+    /// [`AsyncWrite`].
+    ///
+    /// See [`BufWriter`] for details.
+    fn buffered(self) -> BufWriter<Self>
+    where
+        Self: Sized,
+    {
+        BufWriter::new(self)
+    }
+
+    /// Wraps the underlying stream ([`AsyncRead`] + [`AsyncWrite`]) in a [`BufStream`] so that
+    /// small reads and writes are batched to the underlying stream.
+    ///
+    /// See [`BufStream`] for details.
+    fn buffered_duplex(self) -> BufStream<Self>
```

Comment:
In the past, fns that require both `AsyncRead + AsyncWrite` have been defined on the read half.